internal/io/file_system.go

```diff
@@ -14,6 +14,7 @@ type IFileSystem interface {
 	ReadFile(path string) ([]byte, error)
 	Remove(path string) error
 	StatFile(file *os.File) (os.FileInfo, error)
+	IsNotExist(err error) bool
```

Comment:
It's a bit of a nitpick, but if the refactoring is not an issue, I'd suggest changing the name. Either "[Path]DoesNotExist", with Path being optional for clarity. 
```suggestion
	DoesNotExist(err error) bool
```

Reply:
I see no reason to block the PR based on language so I'm approving it. Your choice if you want to change it.

Reply:
Since it's an extension of os.IsNotExist I will keep it as it is, but I agree otherwise!